src/Components/Address/useAddressAutocomplete.ts

```diff
@@ -154,15 +159,26 @@ export const useAddressAutocomplete = (
           : filterSecondarySuggestions(response.suggestions)
 
         setResult(finalSuggestions.slice(0, 5))
+
+        trackEvent({
+          action: ActionType.addressAutoCompletionResult,
+          context_module: ContextModule.ordersShipping,
+          context_page_owner_type: "orders-shipping",
+          context_page_owner_id: contextPageOwnerId,
+          input: search,
+          suggested_addresses_results: finalSuggestions.length,
+        })
```

Comment:
nit: Can we add `as TheCohesionType` types to all of the params we pass to `trackEvent()`? This surfaces errors in our interfaces:
<img width="515" alt="image" src="https://github.com/artsy/force/assets/9088720/30470032-1b1b-4c1c-9728-27c2bc5e9f96">

Reply:
Update: In the case above i think our cohesion schema is actually inconsistent (see the doc comment vs the interface [here](https://github.com/artsy/cohesion/pull/457/files#diff-ee70df3a6cb0738382827014e641e2491d82c625fcd0596f7f1075da3454b724R249-R260)). I'd still like to add the `as TheType` bits to help us catch things like this.

Reply:
is the goal of adding` TheCohesionType` as a type to the params in trackEvent() to enforce type checking to make sure that the values compile as expected?

Reply:
Created the [PR](https://github.com/artsy/cohesion/pull/465)